### PR TITLE
feat(build/k8s): support deploy with Dockerfile

### PR DESCRIPTION
**Pull request by nettoclaudio**
As of https://github.com/tsuru/deploy-agent/commit/701d1f514f7a08da5f5be87f6c54df98d095a099, Deploy Agent v2 supports deploy w/ Dockerfile. Following that changes, this PR implements the support deploy w/ Dockerfile from Tsuru API as wel.

### CODE REVIEW

File: builder/kubernetes/builder.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func (b *kubernetesBuilder) Build(ctx context.Context, prov provision.BuilderDep
 	if !ok {
 		return nil, errors.New("provisioner not supported")
 	}
-	if opts.BuildFromFile {
+	if opts.Dockerfile != "" {
```
Comment (tcarreira) [inline builder/kubernetes/builder.go:36]: does it make sense to notify about this feature with deployV2 here (different error message)?

**Review by tcarreira — APPROVED** [review]

File: builder/kubernetes/builder.go (hunk 1 above)
Comment (nettoclaudio) [reply builder/kubernetes/builder.go:36]: This code is legacy...  it's mostly related with deploy-agent v1.

File: api/deploy_test.go [hunk 2]
```diff
@@ -818,6 +858,14 @@ func (s *DeploySuite) TestPermSchemeForDeploy(c *check.C) {
 			app.DeployOptions{File: ioutil.NopCloser(bytes.NewReader(nil)), Build: true},
 			permission.PermAppDeployBuild,
 		},
+		{
+			app.DeployOptions{Dockerfile: "FROM busybox", File: ioutil.NopCloser(bytes.NewReader(nil))},
+			permission.PermAppDeployDockerfile,
+		},
+		{
```
[the remark this answers is not in the document]
Comment (nettoclaudio) [reply api/deploy_test.go:865]: Ops... it should not set the `File` field :face_exhaling:

[the remark this answers is not in the document]
Comment (nettoclaudio) [reply api/deploy_test.go:865]: Fixed... thanks. :v:

**Comment by tcarreira** [comment]
closes #2569

**Pull request merged**